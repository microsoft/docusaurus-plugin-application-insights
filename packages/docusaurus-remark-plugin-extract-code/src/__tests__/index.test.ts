import path from "path";
import vfile from "to-vfile";
import mdx from "remark-mdx";
import remark from "remark";
import plugin from "../index";

const processFixture = async (name: string,
    options: {
        outputPath?: string;
        langs?: string[];
    }
) => {
    const filePath = path.join(__dirname, "__fixtures__", `${name}.md`);
    const file = await vfile.read(filePath);
    const result = await remark().use(mdx).use(plugin, options).process(file);

    return result.toString();
};

describe("extract-code plugin", () => {
    it("works on snippet file", async () => {
        const result = await processFixture("snippets", { langs: ["js"] });
        expect(result).toMatchSnapshot();
    });
});
