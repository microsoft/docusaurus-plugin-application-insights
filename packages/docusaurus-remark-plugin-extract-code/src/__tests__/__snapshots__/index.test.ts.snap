// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extract-code plugin works on snippet file 1`] = `
"Some snippets

\`\`\`js
let snippet1
\`\`\`

next

\`\`\`js
let snippet2
\`\`\`

And last

\`\`\`js
let snippet3
\`\`\`
"
`;
